pkg/ccl/changefeedccl/testfeed_test.go

```diff
@@ -1041,52 +1058,141 @@ func extractKeyFromJSONValue(isBare bool, wrapped []byte) (key []byte, value []b
-					// the key is recoverable. Extract it out of the value (also removing it
-					// so the output matches the other sinks). Note that this assumes the
-					// format is json, this will have to be fixed once we add format=avro
-					// support to cloud storage.
-					//
-					// TODO(dan): Leave the key in the value if the TestFeed user
-					// specifically requested it.
-					if m.Key, m.Value, err = extractKeyFromJSONValue(c.isBare, m.Value); err != nil {
-						return nil, err
-					}
-					if isNew := c.markSeen(m); !isNew {
-						continue
+		if err != nil {
+			return err
+		}
+
+		// Holds column to its value mapping for a row and the entire thing will be
+		// JSON encoded later.
+		value := make(map[string]interface{})
+
+		// Holds the list of values of primary keys.
+		key := make([]interface{}, 0)
+
+		for k, v := range row {
+
+			switch vv := v.(type) {
+			case []byte:
+				// Currently, for encoding from CRDB data type to Parquet data type, for
+				// any complex structure (that is, other than ints and floats), it is
+				// always a byte array which is the string representation of that datum
+				// (except for arrays, see below). Therefore, if the parquet reader
+				// decodes a column value as a Go native byte array, then we can be sure
+				// that it is the equivalent string representation of the CRDB datum.
+				// Hence, we can use this value to construct the final JSON object which
+				// will be used by assertPayload to compare actual and expected JSON
+				// objects.
+
+				// Ideally there should be no need to convert byte array to string but
+				// JSON encoder will encode byte arrays as base 64 encoded strings.
+				// Hence, we need to convert to string to tell Marshal to decode it as a
+				// string. For every other Go native type, we can use the type as is and
+				// json.Marhsal will work correctly.
+				value[k] = string(vv)
+			case map[string]interface{}:
+				// This is CRDB ARRAY data type (only data type for which we use parquet
+				// LIST logical type. For all other CRDB types, it's either a byte array
+				// or a primitive parquet type). See importer.NewParquetColumn for
+				// details on how CRDB Array is encoded to parquet. (read it before
+				// reading the rest of the comments). Ideally, the parquet reader should
+				// decode the encoded CRDB array datum as a go native list type. But we
+				// use a low level API provided by the vendor which decodes parquet's
+				// LIST logical data type
+				// (https://github.com/apache/parquet-format/blob/master/LogicalTypes.md)
+				// into this weird map structure in Go (It actually makes a lot of sense
+				// why this is done if you understand the parquet LIST logical data
+				// type). A higher level API would convert this map data structure into
+				// go native list type which is what the code below does. This would
+				// probably need to be changed if the parquet vendor is changed.
```

Comment:
I think I understand what you're trying to say here; What I'm not sure
is if this is the result of our exporter code does? Or is it really the library?
Because looking at the library, I can clearly see that we can create NewListCOlumn or NewMapColumn -- and it does not appear in the library, that there is is strange handling of lists as maps.
Perhaps I'm wrong; I would double check that; maybe add a todo comment or some such to revisit this later.

It just looks strange to me.